packages/core/src/integration-tests/index.test.ts

```diff
@@ -405,6 +405,41 @@ describe("Pilot Integration Tests", () => {
         pilot.perform("Action with write error"),
       ).resolves.not.toThrow();
     });
+
+      it("should add error and not add result for next perform step", async () => {
+          let promptParam: string = "";
+          mockPromptHandler.runPrompt
+              .mockResolvedValueOnce(
+                  'throw new Error("Element not found");',
+              )
+              .mockImplementationOnce((prompt, _snapshot) => {
+                  promptParam = prompt;
+                  return Promise.resolve("// No operation");
+              });
+
+          await pilot.perform("Tap on a non-existent button");
+
+          expect(promptParam).toContain("Element not found");
+          expect(promptParam).toMatchSnapshot();
+      });
+
+      it("should add result and not add error for next perform step", async () => {
+          let promptParam: string = "";
+          mockPromptHandler.runPrompt
+              .mockImplementationOnce((prompt, _snapshot) => {
+                  promptParam = prompt;
+                  return Promise.resolve("Operation successful");  // Return a successful result here
+              })
+              .mockImplementationOnce((prompt, _snapshot) => {
+                  promptParam = prompt;
+                  return Promise.resolve("// No operation");  // Return another result on the second call
```

Comment:
If I get the test right - we don't need a second one here, because it will succeed on the first one